### DISCUSSION TITLE
Is it possible to transfer dicussions from Issues to here?

### DISCUSSION THREAD

**Discussion by twMat** [TiddlyWiki/TiddlyWiki5#5301] [General]
Is there some GH feature to move e.g non-actionable issues to this Discussion section? Especially to *force* move, i.e so that moderators can do it.

BTW, is there anyone beyond @Jermolene who has admin rights for this repo?

**Comment by pmario** [comment]
It is possible to move issues to discussions. .. But atm it's only Jeremy, who can do it.

**Comment by twMat** [comment]
>only Jeremy, who can do it.

Then, for practical purposes, one idea for Jeremy might be to "crudely" transfer *all* issues older than, say, 2 years to the Discussion section. As some issues are eventually found to be "wrongly transferred", it can either be alerted for re-instatement or it can be re-created as a new issue.

**Reply by Jermolene** [reply]
Hi @twMat what would be the benefit? There's a lot of good stuff in our old tickets/PRs and I don't think moving things about will make that any more accessible. As things stand, there's not much difference between issues and discussions from my point of view, in particular I still can't distinguish them in the email summaries.

There a lot of open issues, but I think if we think that's a problem we should just use a tool to close them all.

**Comment by twMat** [comment]
@Jermolene - I recall a podcast interview with you where the interviewers were shocked at the number of issues we have, and I suspect newcomers - especially valuable coders - might be deterred from this. Many *discussions*, on the other hand, is probably just a sign of health. Of course, you and I know that our issues list are not pure issues and we also know that the open ended and creative nature of TW itself probably encourages wilder proposals than many other software, but other people probably don't consider this and it more risks looking like the developers are negligent.

**Reply by Jermolene** [reply]
If we're worried about what lots of open issues look like, why wouldn't we just close them? Why move them to be discussions?

**Reply by twMat** [reply]
(???)

Because open issues are not the same as closed issues. IMO closed issues are either resolved or dismissed ones, whereas open issues are still unresolved. My proposal to move them here is because many of the open issues are really discussions and not actionable ones.

The proposal to batch move them comes from the assumption that you (1) are in deed bothered by the number of issues and (2) you want to have sole moderator control and (3) you don't want to go through the isssues one by one to move them. Do I misunderstand?

**Reply by Jermolene** [reply]
But what does an "open" issue from 2015 mean? If it hasn't been commented on for 4 years then it's hard to think of it as being active.

If we closed old issues in bulk we'd do so with a bot that left a message explaining that the ticket had been closed because of elapsed time, and making it clear that people should feel free to reopen it if it is still relevant. I believe that many projects run things like that.

**Reply by saqimtiaz** [reply]
@Jermolene I would favour closing older issues and only potentially moving newer ones that are not actionable but merit further discussion. I am somewhat concerned that as a lot of discussions accumulate, specific ideas/topics will get harder to find. At least Issues get archived in a manner of speaking when closed.

I don't much find it an issue to have too many issues. However it is difficult to identify Issues that have reached a firm proposal for implementation and just need for someone to submit a PR. This holds back contributions to the core, especially from newer developers looking for something to get their feet wet with. Alternatively we could start tagging such issues as well.

**Reply by twMat** [reply]
>But what does an "open" issue from 2015 mean? If it hasn't been commented on for 4 years then it's hard to think of it as being active.

It's not *active* but it is also not resolved. If I look back at my own older+open issues, I think they are still relevant but I don't think anyone would want me to "bump" them every now and then just to keep them "active". And do consider that when you, as competent coders, look back at older issues, you mostly see what you consider *less* interesting stuff because you've typically *solved* the things you care about already.

If (if!) there is a bulk closing I sure hope there's some way to distinguish between those issues and the 1800+ issues that have really been closed, and to do so without opening the individual threads.

**Reply by Jermolene** [reply]
In practice I think "open" vs. "closed" for old issues isn't as meaningful as we might hope. There are lots of issues with useful discussion in that are closed (because the underlying request was actioned), and lots of issues that are open that could in fact be closed on our usual criteria.

>  If I look back at my own older+open issues, I think they are still relevant but I don't think anyone would want me to "bump" them every now and then just to keep them "active"

I think in fact, "bumping" old issues is the right way for a contributor to bring attention to an old issue that they feel is relevant. The "open" status is clearly not enough; we've too many to review. Plus we tend to err on the side of keeping issues open, to avoid it feeling like we're slamming the door in a would-be contributors face.

Perhaps we might say that a "closed" issue does make a positive indication that something has been decided, but an "open" issue doesn't mean anything by itself.

> If (if!) there is a bulk closing I sure hope there's some way to distinguish between those issues and the 1800+ issues that have really been closed, and to do so without opening the individual threads

I think we'd need to apply a label in that situation (there would automatically be a note in the thread of the issue itself). But I'd question the utility of that label; what would it be used for?

**Comment by twMat** [comment]
> [...] lots of issues that are open that could in fact be closed on our usual criteria. [...] an "open" issue doesn't mean anything by itself.

"Open" is surely intended to mean "unresolved" so we're just not using it the right way. But I agree it would be better if there were further indicators...

But, even if all open issues were closed; then what? Would "open mean something by itself" then?

> [...] But I'd question the utility of that label; what would it be used for?

It would not be for you. Actually, as a super coder and the "owner" of this project, you are the very last person it would be of value for.
Instead it would be for people like me or other non-super contributors because we are likely over represented with the open issues. We/I are typically just waiting and hoping that some of you big boys will one day agree with our concern (or that I/we ourselves change opinion of the issue and deem it dismissable). So a distinguishing label is primarily for the issue-poster and those who care about the issue so that these issues can be located among the resolved ones. I try to go through my issues from time to time.

In contrast, a resolved or fully dismissed issue, be it mine or someone elses, could more or less be erased completely as far as I'm concerned.

OR maybe you have something better in mind? How would I find, say, [this oldest, open, issue of mine](https://github.com/Jermolene/TiddlyWiki5/issues/1112) among the closed ones?

**Reply by Jermolene** [reply]
> It would not be for you. Actually, as a super coder and the "owner" of this project, you are the very last person it would be of value for.
> Instead it would be for people like me or other non-super contributors because we are likely over represented with the open issues. We/I are typically just waiting and hoping that some of you big boys will one day agree with our concern (or that I/we ourselves change opinion of the issue and deem it dismissable). So a distinguishing label is primarily for the issue-poster and those who care about the issue so that these issues can be located among the resolved ones. I try to go through my issues from time to time.

I don't disagree that a label for issues that have been closed by a bot could be useful. However, I don't think the presence of a forgotten open issue is ever going to catalyse the action that you seek. Speaking for myself, I don't think I will never, ever have the time or motivation for a systematic review of the open issues. If you think an old issue is relevant for discussion or action you can just bump the issue, which is guaranteed to bring the attention of your fellow collaborators.

> In contrast, a resolved or fully dismissed issue, be it mine or someone elses, could more or less be erased completely as far as I'm concerned.

I still maintain that closed issues are useful. Sometimes I spend a long time composing a careful answer explaining why I think something may not work, and I'd hate for that work to be lost. That's the trouble with issues: they generally take much, much more time to process than they do to create.

> OR maybe you have something better in mind? How would I find, say, [this oldest, open, issue of mine](https://github.com/Jermolene/TiddlyWiki5/issues/1112) among the closed ones?

The core of our issue here is that I am not sure that having lots of issues like that with the "open" status is of much value to the project. The idea you're proposing is a good one, and might be worth pursuing, but this feels like the wrong place for it. Why would we want 100s of open issues that describe good ideas that might be worth pursuing? They're very hard to spot amongst the 100s of ideas that aren't worth pursuing. Arguably, the really important open issues are the bugs that remain unfixed, but again it's very hard to spot them if we use the "open" status as a sort of long term wishlist of features that people would like to see.

To me "open" should mean under active discussion, and it seems reasonable to follow other projects in automatically closing open issues after a period of inactivity to give us some hope of being able to see the wood for the trees.

**Comment by pmario** [comment]
>If we closed old issues in bulk we'd do so with a bot that left a message explaining that the ticket had been closed because of elapsed time, and making it clear that people should feel free to reopen it if it is still relevant. 

I can not reopen an issue, if it isn't mine OR if you (Jeremy) closed it. If you close it, it's closed!

>I believe that many projects run things like that.

Because there was no other possibility. IMO there is a reason, why there is a new "discussion" area.

**Comment by pmario** [comment]
For me personally an "issue" and a "discussions" are fundamentally different things.

An issue has 3 states: 

 - open
 - closed and fixed
 - closed and wontfix

If they are open, it means the project has a problem. The more open issues a project has, the more problems it has!

If they are closed, they are "archived". 

For me there is absolutely no need to "re-read" closed issues. I sometimes find them in search results, but if an issue has been closed by Jeremy, it doesn't make much sense to reopen it. 

----------

Discussions have only 1 state: "ongoing".

**Reply by twMat** [reply]
What Mario just wrote *exactly* reflects my view on this.

**Comment by morosanuae** [comment]
It looks like this project needs a full-time administrator/manager, otherwise, even if we start clean, it will become a mess again very soon IMO. The lack of time, motivation and workforce simply cannot be substituted.
